sale_credit_point/models/point_history.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+
+
+class PointHistory(models.Model):
+    _name = "credit.point.history"
+    _rec_name = "partner_id"
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Partner",
+        required=True,
+    )
+    operation = fields.Selection(
+        string="Operation",
+        selection="_get_operation_selection",
+        required=True,
+    )
+    amount = fields.Monetary(
+        string='Amount',
+        currency_field='credit_point_currency_id',
+        readonly=True,
+        default=0,
+        required=True,
+    )
+    credit_point_currency_id = fields.Many2one(
+        related="partner_id.credit_point_currency_id",
+        readonly=True,
+    )
+    execution_datetime = fields.Datetime(
+        string="Execution time",
+        default=lambda self: self._default_execution_datetime(),
+        required=True,
+    )
+    year = fields.Char(
```

Comment:
I don't think you need this either. You can [select dates by year](https://stackoverflow.com/questions/35704412/postgresql-query-current-year).
If you decide to keep it, make it an index.